src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,177 @@
+     *
+     * @param input user input.
+     * @return a Event based on user input.
+     * @throws DukeException when user input does not contain
+     *                       the description of the Event after the command,
+     *                       the date and time of the Event after '/at'
+     *                       or when the date and time is not in the correct format.
+     */
+    public Event parseEvent(String input) throws DukeException {
+        String[] arr = input.split(" ");
+        String[] arr3 = input.split(" /at ");
+        if (arr.length < 2 || arr3[0].split(" ").length < 2) {
+            throw new DukeException("OOPS!!! The description of an event cannot be empty.");
+        }
+        if (arr3.length < 2) {
+            throw new DukeException("OOPS!!! You did not set the date/time of the event using '/at'.");
+        }
+        LocalDateTime eventDate;
+        try {
+            eventDate = LocalDateTime.parse(arr3[1], INPUT_DATETIME_FORMAT);
+        } catch (DateTimeParseException dateTimeParseException) {
+            throw new DukeException("OOPS!!! Please input date and time in the following format: dd-MM-yyyy HHmm.");
+        }
+        return new Event(arr3[0].substring(6), eventDate);
+    }
+
+    /**
+     * Returns a valid int index of the Task
+     * to be marked as done in the TaskList.
+     *
+     * @param input    user input.
+     * @param taskList list of Tasks that contains the Task to be marked as done.
+     * @return the index of the Task.
+     * @throws DukeException when user input does not contain an int value after the command
+     *                       or when the index provided does not correspond to any Task in the TaskList.
+     */
+    public int parseDone(String input, TaskList taskList) throws DukeException {
+        ArrayList<Task> tasks = taskList.getTasks();
+        String[] arr = input.split(" ");
+        if (arr.length < 2) {
+            throw new DukeException("OOPS!!! You did not enter the number corresponding to the task.");
+        }
+        try {
+            int argument = Integer.parseInt(arr[1]);
+            if (argument < 1 || argument > tasks.size()) {
+                throw new DukeException("OOPS!!! The numbered task does not exist.");
+            }
+            return argument - 1;
+        } catch (NumberFormatException numberFormatException) {
+            throw new DukeException("OOPS!!! You did not enter the number corresponding to the task.");
+        }
+    }
+
+    /**
+     * Returns a valid int index of the Task
+     * to be deleted from the TaskList.
+     *
+     * @param input    user input.
+     * @param taskList list of Tasks that contains the Task to be deleted.
+     * @return the index of the Task.
+     * @throws DukeException when user input does not contain an int value after the command
+     *                       or when the index provided does not correspond to any Task in the TaskList.
```

Comment:
i am not sure if it's a glitch on github's part or ur IDE altered the spacings for ur javaDocs but do check it out i am sure this isnt intentional hahaha